### PR TITLE
profiler: fix newlines breaking up log messages

### PR DESCRIPTION
**Pull request by felixge**
Also remove a `printf` statement that seems like it was left in the code by accident.

### CODE REVIEW

File: profiler/profiler.go [hunk 1]
```diff
@@ -160,8 +160,7 @@ func (p *profiler) collect(ticker <-chan time.Time) {
 			for t := range p.cfg.types {
 				prof, err := p.runProfile(t)
 				if err != nil {
-					fmt.Printf("error: %v\n", err)
```
Comment (felixge) [inline profiler/profiler.go:163]: @pmbauer I suspect you left this in 634b2317 by accident?

**Review by gbbr — APPROVED** [review]

**Comment by gbbr** [comment]
You'll have to rebase, which'll dismiss the review, but will happily re-approve...

**Comment by felixge** [comment]
@gbbr just rebased it, 👍

**Review by gbbr — APPROVED** [review]

File: profiler/profiler.go (hunk 1 above)
Comment (pmbauer) [reply profiler/profiler.go:163]: looks like it

Comment (felixge) [reply profiler/profiler.go:163]: @pmbauer ok, thanks for confirming, just wanted to make sure it wasn't intentional. I mean it seems like this is definitely an error path that'd be very unexpected for us to hit.

**Pull request merged**